astropy/coordinates/tests/test_astrometric_transformations.py

```diff
@@ -0,0 +1,169 @@
+# -*- coding: utf-8 -*-
+
+# -*- coding: utf-8 -*-
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+import numpy as np
+
+from ... import units as u
+from ..builtin_frames import (ICRS, AstrometricICRS)
+from .. import SkyCoord
+from ...tests.helper import (pytest, quantity_allclose as allclose,
+                             assert_quantity_allclose as assert_allclose)
+
+
+@pytest.mark.parametrize("inradec,expecteddradec, tolsep", [
+    ((45, 45)*u.deg, (0, 0)*u.deg, .001*u.arcsec),
+    ((45, 0)*u.deg, (0, -45)*u.deg, .001*u.arcsec),
+    ((45, 90)*u.deg, (0, 45)*u.deg, .001*u.arcsec),
+    ((46, 45)*u.deg, (1*np.cos(45*u.deg), 0)*u.deg, 16*u.arcsec),
+])
+def test_astrometric(inradec, expecteddradec, tolsep, originradec=(45, 45)*u.deg):
+    origin = ICRS(*originradec)
+    astrometric_frame = AstrometricICRS(origin=origin)
+
+    skycoord = SkyCoord(*inradec, frame=ICRS)
+    skycoord_inaf = skycoord.transform_to(astrometric_frame)
+    assert hasattr(skycoord_inaf, 'dra')
+    assert hasattr(skycoord_inaf, 'ddec')
+    expected = SkyCoord(*expecteddradec, frame=astrometric_frame)
+
+    assert skycoord_inaf.separation(expected) < tolsep
+
+
+def test_astrometric_functional_ra():
+    #Setup
+    input_ra = np.linspace(0,360,10)
+    input_dec = np.linspace(-90,90,10)
+    input_ra_rad = np.deg2rad(input_ra)
+    input_dec_rad = np.deg2rad(input_dec)
+    icrs_coord = ICRS(ra = input_ra*u.deg,
+                      dec = input_dec*u.deg,
+                      distance=1.*u.kpc)
+    #RA rotations
+
+    for ra in np.linspace(0,360,24):
+        # expected rotation
+        expected = ICRS(ra=np.linspace(0-ra,360-ra,10)*u.deg,
+                        dec=np.linspace(-90,90,10)*u.deg,
+                        distance=1.*u.kpc)
+        expected_xyz = expected.cartesian.xyz
+
+        # actual transformation to the frame
+        astrometric_frame = AstrometricICRS(origin=ICRS(ra*u.deg, 0*u.deg))
+        actual = icrs_coord.transform_to(astrometric_frame)
+        actual_xyz = actual.cartesian.xyz
+
+        # back to ICRS
+        roundtrip = actual.transform_to(ICRS)
+        roundtrip_xyz = roundtrip.cartesian.xyz
+
+        # Verify
+        assert_allclose(actual_xyz.to(u.kpc), expected_xyz.to(u.kpc), atol=1E-5*u.kpc)
+        #assert_allclose(actual_xyz.to(u.kpc), roundtrip_xyz.to(u.kpc), atol=1E-5*u.kpc)
```

Comment:
Oops, yeah, I think that's supposed to be removed (this was from the #4524 branch and I didn't look through it terribly carefully...)